Make it build with ghc-9.6

erikd opened the pull request:
# Changelog

```yaml
- description: Make it build with ghc-9.6
  compatibility: no-api-changes
  type: maintenance
```

edolstra commented:
The crash should be fixed by 8c5bbba722b8621b7ff0c781edd2c3e6a20da67f. I also updated the PR to use rust-cardano master.

erikd committed:
Make it build with ghc-9.6

erikd committed:
Update Nix flake lock file

andreabedini commented on cabal.project:
```diff
@@ -37,6 +37,19 @@ test-show-details: direct
 write-ghc-environment-files: always
 
+-- https://github.com/obsidiansystems/dependent-sum-template/issues/5
+if impl(ghc >= 9.2)
```
Does this mean this is necessary also for ghc-9.4?

andreabedini committed:
Avoid haskell.nix broken commit in nix flake lockfile

Using

  github:input-output-hk/haskell.nix/50e7b671ca73ea9ab350d7f18ba434bf8393051a

until https://github.com/input-output-hk/haskell.nix/pull/2003 gets merged.

andreabedini committed:
Bump haskell.nix again

erikd committed:
Release to CHaP

michaelpj commented on cabal.project:
```diff
@@ -37,6 +37,19 @@ test-show-details: direct
+-- https://github.com/obsidiansystems/dependent-sum-template/issues/5
+if impl(ghc >= 9.2)
+  constraints :
+      dependent-sum-template < 0.1.2
```
Why is this necessary in the build? Please leave a comment otherwise we'll never know when we can remove it.

I know we had a problem here because `plutus-core` pulled in the newer `dependent-sum-template` but I think we can fix that much more nicely by revising `plutus-core` in CHaP (which I've done) rather than propagating such conditionals everywhere.

andreabedini replied:
I think it is tricky to understand what can be fixed and where it should be fixed. Maybe some tooling or simply documentation could help? E.g. how could Erik realise this could have been fixed in plutus-core?

erikd replied:
It was a dependency from (I think) plutus-core.

erikd replied:
I was actually able to remove that conditional.

michaelpj replied:
@andreabedini yeah, I think this is not so easy with plain cabal, I don't think it gives you the stack of things that needed the library that doesn't build. I guess I usually get that info from Nix :shrug: 

Possibly building with `--constraint dependent-sum-template < 0` will get the solver to tell you what requires it.

I know I keep harping on about this, but it is _so_ much easier to maintain this stuff if we can fix bounds in CHaP once and for all rather than having a thin spread of conditional logic across our many project files.
